### PR TITLE
SL-836: upgrade types

### PR DESCRIPTION
**Pull request by chohmann**

### CODE REVIEW

File: packages/http/src/mocker/HttpMocker.ts [hunk 1]
```diff
@@ -45,8 +45,8 @@ export class HttpMocker
 
     // preparing response body
     let body;
-    if (negotiationResult.example) {
-      body = negotiationResult.example.value;
+    if (negotiationResult.example && (negotiationResult.example as INodeExample).value) {
```
Comment (chris-miaskowski) [inline packages/http/src/mocker/HttpMocker.ts]: Do we allow `value` to be an empty string for instance?

Comment (chohmann) [reply packages/http/src/mocker/HttpMocker.ts]: Good catch! I changed to check specifically for undefined.

File: packages/http/src/mocker/HttpMocker.ts [hunk 2]
```diff
@@ -45,8 +45,11 @@ export class HttpMocker
 
     // preparing response body
     let body;
-    if (negotiationResult.example) {
-      body = negotiationResult.example.value;
+    if (
+      negotiationResult.example &&
+      (negotiationResult.example as INodeExample).value !== undefined
+    ) {
+      body = (negotiationResult.example as INodeExample).value;
```
Comment (pytlesk4) [inline packages/http/src/mocker/HttpMocker.ts]:
```ts
  let body;
  const example = negotiationResult.example as INodeExample;
    if (example && example.value !== undefined ) {
      body = example.value;
```

File: packages/http/src/router/types.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { IHttpOperation } from '@stoplight/types/http';
+import { IHttpOperation } from '@stoplight/types/http-spec';
```
Comment (pytlesk4) [inline packages/http/src/router/types.ts]:
```ts
import { IHttpOperation } from '@stoplight/types';
```

File: packages/http/src/mocker/negotiator/types.ts [hunk 4]
```diff
@@ -1,9 +1,9 @@
-import { IExample } from '@stoplight/types/http';
-import { ISchema } from '@stoplight/types/schema';
+import { INodeExample, INodeExternalExample } from '@stoplight/types/graph';
```
Comment (pytlesk4) [inline packages/http/src/mocker/negotiator/types.ts]: import { INodeExample, INodeExternalExample, ISchema } from '@stoplight/types';

File: packages/http/src/mocker/negotiator/types.ts [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/mocker/negotiator/types.ts]: Delete this line

File: packages/http/src/mocker/__tests__/HttpMocker.spec.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { IHttpOperation } from '@stoplight/types/http';
+import { IHttpOperation } from '@stoplight/types/http-spec';
```
Comment (pytlesk4) [inline packages/http/src/mocker/__tests__/HttpMocker.spec.ts]: import { IHttpOperation } from '@stoplight/types';

File: packages/http/src/mocker/__tests__/functional.spec.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/mocker/__tests__/functional.spec.ts]:
```ts
import { ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/deserializers/style/deepObject.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 import { HttpParamStyles } from '@stoplight/types';
```
Comment (pytlesk4) [inline packages/http/src/validator/deserializers/style/deepObject.ts]:
```ts
import { HttpParamStyles, ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/deserializers/style/deepObject.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
...
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/deserializers/style/deepObject.ts]: delete

File: packages/http/src/validator/deserializers/style/delimited.ts [hunk 10]
```diff
@@ -1,5 +1,5 @@
 import { HttpParamStyles } from '@stoplight/types';
```
Comment (pytlesk4) [inline packages/http/src/validator/deserializers/style/delimited.ts]:
```ts
import { HttpParamStyles, ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/deserializers/style/delimited.ts [hunk 11]
```diff
@@ -1,5 +1,5 @@
...
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/deserializers/style/delimited.ts]: delete

File: packages/http/src/validator/deserializers/style/form.ts [hunk 12]
```diff
@@ -1,5 +1,5 @@
 import { HttpParamStyles } from '@stoplight/types';
```
Comment (pytlesk4) [inline packages/http/src/validator/deserializers/style/form.ts]:
```ts
import { HttpParamStyles, ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/deserializers/style/form.ts [hunk 13]
```diff
@@ -1,5 +1,5 @@
...
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/deserializers/style/form.ts]: delete

File: packages/http/src/validator/deserializers/style/simple.ts [hunk 14]
```diff
@@ -1,5 +1,5 @@
 import { HttpParamStyles } from '@stoplight/types';
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/deserializers/style/simple.ts]:
```ts
import { HttpParamStyles, ISchema } from '@stoplight/types';
```

Comment (pytlesk4) [reply packages/http/src/validator/deserializers/style/simple.ts]: and delete

File: packages/http/src/validator/deserializers/types.ts [hunk 15]
```diff
@@ -1,5 +1,5 @@
 import { HttpParamStyles } from '@stoplight/types';
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/deserializers/types.ts]: delete

File: packages/http/src/validator/deserializers/types.ts [hunk 16]
```diff
@@ -1,5 +1,5 @@
 import { HttpParamStyles } from '@stoplight/types';
```
Comment (pytlesk4) [inline packages/http/src/validator/deserializers/types.ts]:
```ts
import { HttpParamStyles, ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/index.ts [hunk 17]
```diff
@@ -49,7 +49,7 @@ export class HttpValidator
         results,
```
Comment (pytlesk4) [inline packages/http/src/validator/index.ts]:
```ts
// Replace resource.request in this function with request
const { request } = resource;

if (config.body) {
  const { body } = input;
  results.push(this.bodyValidator.validate(
    body,
    (request && request.body && request.body.contents) || [],
    mediaType
  ))
}
```

File: packages/http/src/validator/utils/http.ts [hunk 18]
```diff
@@ -1,4 +1,4 @@
-import { IHttpContent } from '@stoplight/types/http';
+import { IHttpContent } from '@stoplight/types/http-spec';
```
Comment (pytlesk4) [inline packages/http/src/validator/utils/http.ts]:
```ts
import { IHttpContent } from '@stoplight/types';
```

File: packages/http/src/mocker/__tests__/functional.spec.ts (hunk 7 above)
Comment (chohmann) [reply packages/http/src/mocker/__tests__/functional.spec.ts]: Done.

File: packages/http/src/mocker/__tests__/HttpMocker.spec.ts (hunk 6 above)
Comment (chohmann) [reply packages/http/src/mocker/__tests__/HttpMocker.spec.ts]: Done

File: packages/http/src/mocker/negotiator/types.ts (hunk 5 above)
Comment (chohmann) [reply packages/http/src/mocker/negotiator/types.ts]: Done

File: packages/http/src/mocker/negotiator/types.ts (hunk 4 above)
Comment (chohmann) [reply packages/http/src/mocker/negotiator/types.ts]: Done

File: packages/http/src/router/types.ts (hunk 3 above)
Comment (chohmann) [reply packages/http/src/router/types.ts]: Done

File: packages/http/src/mocker/HttpMocker.ts (hunk 2 above)
Comment (chohmann) [reply packages/http/src/mocker/HttpMocker.ts]: Done

File: packages/http/src/validator/utils/spec.ts [hunk 19]
```diff
@@ -1,4 +1,4 @@
-import { IHttpResponse as IHttpResponseSpec } from '@stoplight/types/http';
+import { IHttpOperationResponse as IHttpResponseSpec } from '@stoplight/types/http-spec';
```
Comment (pytlesk4) [inline packages/http/src/validator/utils/spec.ts]: Need to update errors after changing the import. I am concerned that changing the import is confusing, using findResponseSpec that returns an array of IHttpResponseSpec[] looks like a new type when it isn't.

Also, suggest changing the name of findResponseSpec to findResponseOperation?
Also, suggest changing responseSpecs: IHttpOperationResponse[] to responseOperations: ...
```ts
import { IHttpOperationResponse } from '@stoplight/types';
```

File: packages/http/src/validator/validators/__tests__/body.spec.ts [hunk 20]
```diff
@@ -1,5 +1,5 @@
 import { ValidationSeverity } from '@stoplight/prism-core';
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/validators/__tests__/body.spec.ts]:
```ts
import { ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/validators/__tests__/headers.spec.ts [hunk 21]
```diff
@@ -1,5 +1,5 @@
 import { HttpParamStyles } from '@stoplight/types';
```
Comment (pytlesk4) [inline packages/http/src/validator/validators/__tests__/headers.spec.ts]:
```ts
import { HttpParamStyles, ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/validators/__tests__/headers.spec.ts [hunk 22]
```diff
@@ -1,5 +1,5 @@
...
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/validators/__tests__/headers.spec.ts]: delete

File: packages/http/src/validator/validators/__tests__/query.spec.ts [hunk 23]
```diff
@@ -1,5 +1,5 @@
 import { HttpParamStyles } from '@stoplight/types';
```
Comment (pytlesk4) [inline packages/http/src/validator/validators/__tests__/query.spec.ts]:
```ts
import { HttpParamStyles, ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/validators/__tests__/query.spec.ts [hunk 24]
```diff
@@ -1,5 +1,5 @@
...
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/validators/__tests__/query.spec.ts]: delete

File: packages/http/src/validator/validators/__tests__/registry.spec.ts [hunk 25]
```diff
@@ -1,4 +1,4 @@
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/validators/__tests__/registry.spec.ts]:
```ts
import { ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/validators/body.ts [hunk 26]
```diff
@@ -1,5 +1,5 @@
 import { IValidation } from '@stoplight/prism-core';
-import { IHttpContent } from '@stoplight/types/http';
+import { IHttpContent } from '@stoplight/types/http-spec';
```
Comment (pytlesk4) [inline packages/http/src/validator/validators/body.ts]:
```ts
import { IHttpContent } from '@stoplight/types';
```

File: packages/http/src/validator/validators/registry.ts [hunk 27]
```diff
@@ -1,5 +1,5 @@
 import { IValidation } from '@stoplight/prism-core';
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/validators/registry.ts]:
```ts
import { ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/validators/schema.ts [hunk 28]
```diff
@@ -1,5 +1,5 @@
 import { IValidation, ValidationSeverity } from '@stoplight/prism-core';
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/validators/schema.ts]:
```ts
import { ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/validators/types.ts [hunk 29]
```diff
@@ -1,5 +1,5 @@
 import { IValidation } from '@stoplight/prism-core';
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/validators/types.ts]:
```ts
import { ISchema } from '@stoplight/types';
```

File: packages/http/src/validator/validators/utils.ts [hunk 30]
```diff
@@ -1,5 +1,5 @@
 import { IValidation, ValidationSeverity } from '@stoplight/prism-core';
-import { ISchema } from '@stoplight/types/schema';
+import { ISchema } from '@stoplight/types/schemas';
```
Comment (pytlesk4) [inline packages/http/src/validator/validators/utils.ts]:
```ts
import { ISchema } from '@stoplight/types';
```

**Comment by pytlesk4** [comment]
Looks good! Just minor changes and clean up requested!

File: packages/http/src/validator/deserializers/style/deepObject.ts (hunk 9 above)
Comment (chohmann) [reply packages/http/src/validator/deserializers/style/deepObject.ts]: Done

File: packages/http/src/validator/deserializers/style/delimited.ts (hunk 10 above)
Comment (chohmann) [reply packages/http/src/validator/deserializers/style/delimited.ts]: Done

File: packages/http/src/validator/deserializers/style/delimited.ts (hunk 11 above)
Comment (chohmann) [reply packages/http/src/validator/deserializers/style/delimited.ts]: Done

File: packages/http/src/validator/deserializers/style/form.ts (hunk 12 above)
Comment (chohmann) [reply packages/http/src/validator/deserializers/style/form.ts]: Done

File: packages/http/src/validator/deserializers/style/form.ts (hunk 13 above)
Comment (chohmann) [reply packages/http/src/validator/deserializers/style/form.ts]: Done

File: packages/http/src/validator/deserializers/style/simple.ts (hunk 14 above)
Comment (chohmann) [reply packages/http/src/validator/deserializers/style/simple.ts]: Done

File: packages/http/src/validator/deserializers/types.ts (hunk 15 above)
Comment (chohmann) [reply packages/http/src/validator/deserializers/types.ts]: Done

File: packages/http/src/validator/deserializers/types.ts (hunk 16 above)
Comment (chohmann) [reply packages/http/src/validator/deserializers/types.ts]: Done

File: packages/http/src/validator/index.ts (hunk 17 above)
Comment (chohmann) [reply packages/http/src/validator/index.ts]: Done

File: packages/http/src/validator/utils/http.ts (hunk 18 above)
Comment (chohmann) [reply packages/http/src/validator/utils/http.ts]: Done

File: packages/http/src/validator/utils/spec.ts (hunk 19 above)
Comment (chohmann) [reply packages/http/src/validator/utils/spec.ts]: Done

File: packages/http/src/validator/validators/__tests__/body.spec.ts (hunk 20 above)
Comment (chohmann) [reply packages/http/src/validator/validators/__tests__/body.spec.ts]: Done

File: packages/http/src/validator/validators/__tests__/headers.spec.ts (hunk 21 above)
Comment (chohmann) [reply packages/http/src/validator/validators/__tests__/headers.spec.ts]: Done

File: packages/http/src/validator/validators/__tests__/headers.spec.ts (hunk 22 above)
Comment (chohmann) [reply packages/http/src/validator/validators/__tests__/headers.spec.ts]: Done

File: packages/http/src/validator/validators/__tests__/query.spec.ts (hunk 23 above)
Comment (chohmann) [reply packages/http/src/validator/validators/__tests__/query.spec.ts]: Done

File: packages/http/src/validator/validators/__tests__/query.spec.ts (hunk 24 above)
Comment (chohmann) [reply packages/http/src/validator/validators/__tests__/query.spec.ts]: Done

File: packages/http/src/validator/validators/__tests__/registry.spec.ts (hunk 25 above)
Comment (chohmann) [reply packages/http/src/validator/validators/__tests__/registry.spec.ts]: Done

File: packages/http/src/validator/validators/body.ts (hunk 26 above)
Comment (chohmann) [reply packages/http/src/validator/validators/body.ts]: Done

File: packages/http/src/validator/validators/registry.ts (hunk 27 above)
Comment (chohmann) [reply packages/http/src/validator/validators/registry.ts]: Done

File: packages/http/src/validator/validators/schema.ts (hunk 28 above)
Comment (chohmann) [reply packages/http/src/validator/validators/schema.ts]: Done

File: packages/http/src/validator/validators/types.ts (hunk 29 above)
Comment (chohmann) [reply packages/http/src/validator/validators/types.ts]: Done

File: packages/http/src/validator/validators/utils.ts (hunk 30 above)
Comment (chohmann) [reply packages/http/src/validator/validators/utils.ts]: Done

File: packages/http/src/validator/index.ts [hunk 31]
```diff
@@ -44,37 +44,35 @@ export class HttpValidator
...
     }
 
     if (config.headers) {
-      Array.prototype.push.apply(
-        results,
-        this.headersValidator.validate(
+      this.headersValidator
+        .validate(
           input.headers || {},
           (resource.request && resource.request.headers) || [],
```
Comment (pytlesk4) [inline packages/http/src/validator/index.ts]:
```ts
(request && request.headers) || []
```

File: packages/http/src/validator/index.ts [hunk 32]
```diff
@@ -44,37 +44,35 @@ export class HttpValidator
...
     }
 
     if (config.query) {
-      Array.prototype.push.apply(
-        results,
-        this.queryValidator.validate(
+      this.queryValidator
+        .validate(
           input.url.query || {},
           (resource.request && resource.request.query) || [],
```
Comment (pytlesk4) [inline packages/http/src/validator/index.ts]:
```ts
(request && request.query) || []
```

File: packages/http/src/validator/index.ts (hunk 31 above)
Comment (chohmann) [reply packages/http/src/validator/index.ts]: Done

File: packages/http/src/validator/index.ts (hunk 32 above)
Comment (chohmann) [reply packages/http/src/validator/index.ts]: Done.

**Pull request merged**